cli/pkg/timescaledb_secrets/create.go

```diff
@@ -0,0 +1,316 @@
+)
+
+type TSDBSecretsInfo struct {
+	ReleaseName    string
+	Namespace      string
+	EnableS3Backup bool
+	TlsCert        []byte
+	TlsKey         []byte
+}
+
+func (t *TSDBSecretsInfo) CreateTimescaleDBSecrets() error {
+	// Previous helm install used to create namespace if it doesn't exist
+	// but as we are creating secrets prior to deploying tobs. We are verifying
+	// namespace if doesn't create one.
+	err := k8s.CreateNamespaceIfNotExists(t.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if t.TlsCert == nil && t.TlsKey == nil {
+		err = t.createTimescaleDBCertificates()
+		if err != nil {
+			return err
+		}
+	} else {
+		fmt.Println("Using the provided TLS certificates for TimescaleDB.")
+	}
+
+	err = t.createTimescaleDBCredentials()
+	if err != nil {
+		return err
+	}
+
+	if t.EnableS3Backup {
+		err = t.createS3BackupForTimescaleDB()
+		if err != nil {
+			return err
+		}
+	} else {
+		fmt.Println("skipping to create TimescaleDB s3 backup secret as backup option is disabled.")
+	}
+
+	return nil
+}
+
+// randomPassword generates a random alphanumeric password of a given length.
+// Compatible with passwords generated by the postgres operator
+func randomPassword(n int) ([]byte, error) {
+	var passwordChars = []byte("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
+	b := make([]byte, n)
+	_, err := crand.Read(b)
+	if err != nil {
+		return b, fmt.Errorf("failed to generate random password %v", err)
+	}
+	for i := 0; i < n; i++ {
+		b[i] = passwordChars[b[i]/5]
+	}
+	return b, nil
+}
+
+func (t *TSDBSecretsInfo) createTimescaleDBCredentials() error {
+	secretName := t.ReleaseName + "-credentials"
+	exists, err := k8s.CheckSecretExists(secretName, t.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if exists {
+		fmt.Printf("using existing %s secret\n", secretName)
+		return nil
+	}
+
+	fmt.Printf("Creating TimescaleDB %s secret\n", secretName)
+	repPass, err := randomPassword(10)
+	if err != nil {
+		return err
+	}
+	adminPass, err := randomPassword(10)
+	if err != nil {
+		return err
+	}
+	superUserPass, err := randomPassword(10)
+	if err != nil {
+		return err
+	}
+	sec := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: t.Namespace,
+		},
+		Data: map[string][]byte{
+			"PATRONI_REPLICATION_PASSWORD": repPass,
+			"PATRONI_admin_PASSWORD":       adminPass,
+			"PATRONI_SUPERUSER_PASSWORD":   superUserPass,
+		},
+		Type: "Opaque",
+	}
+
+	return k8s.CreateSecret(sec)
+}
+
+func generateCerts() ([]byte, []byte, error) {
+	priv, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate rsa key: %v", err)
+	}
+
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to generate serial number: %v", err)
+	}
+
+	template := x509.Certificate{
+		SerialNumber:          serialNumber,
```

Comment:
let's add 	following the kubernetes ingress controller https://i.stack.imgur.com/iyjjx.png	
Subject: pkix.Name{
			CommonName:   "Tobs TimescaleDB Fake Certificate",
			Organization: []string{"Acme Co"},
},